cpp/ql/lib/semmle/code/cpp/controlflow/SSAUtils.qll

```diff
@@ -153,7 +153,7 @@ library class SSAHelper extends int {
    * Modern Compiler Implementation by Andrew Appel.
    */
   private predicate frontier_phi_node(StackVariable v, BasicBlock b) {
-    exists(BasicBlock x | dominanceFrontier(x, b) and ssa_defn_rec(v, x)) and
+    exists(BasicBlock x | dominanceFrontier(pragma[only_bind_into](x), b) and ssa_defn_rec(v, x)) and
```

Comment:
What is this meant to accomplish? It doesn't really push the optimiser towards the good join-order as presented in the PR description. On the contrary it actually looks more like it attempts to prevent the good join-order, but then ends up working accidentally, since this pragma doesn't actually force anything.

Reply:
Thanks! Yeah, that join-order fix clearly wasn't correct. Hopefully the one in c1af8b93c28474ee67f52dbfdfb98e7cc132c8af is better.